packages/cozy-harvest-lib/src/models/cipherUtils.js

```diff
@@ -113,7 +113,9 @@ export const createOrUpdateCipher = async (
     })
   }
 
-  await shareCipherWithCozy(vaultClient, cipher.id)
+  if (cipher) {
+    await shareCipherWithCozy(vaultClient, cipher.id)
+  }
```

Comment:
Is it possible to have no `cipher` here? How since we create it or update it. Is it to prevent an error during those 2 methods? 

```


  if (cipherId) {
    cipher = await updateCipher(vaultClient, cipherId, { login, password })
  } else {
    cipher = await createCipher(vaultClient, {
      konnector,
      login,
      password
    })
  }

if(cipher)... 
```

Reply:
If the vault is unlocked, no cipher can be created. I have a problem here though since updateCipher does check if the vault is locked or not (contrary to createCipher). I am moving the check in this function and adding tests. Thanks for the catch.